### PR TITLE
CON-4696

### PR DESCRIPTION
**Pull request by SimonBaeumer**

### CODE REVIEW

File: Components/CategoryQuery/Sw41Query.php [hunk 1]
```diff
@@ -7,40 +7,62 @@
...
 
-        $query = $builder->getQuery();
-        $query->setHydrationMode($query::HYDRATE_OBJECT);
+    /**
+     * @return \Shopware\Models\Article\Repository
+     */
+    protected function getArticleRepository()
+    {
+        $repository = $this->manager->getRepository(
+            'Shopware\Models\Article\Article'
```
Comment (sbofirov) [inline Components/CategoryQuery/Sw41Query.php]: you can use ```Article::class```

File: Components/CategoryQuery/Sw41Query.php [hunk 2]
```diff
@@ -7,40 +7,62 @@
...
+    }
 
-        return $categories;
+    /**
+     * @return \Shopware\Models\Category\Repository
+     */
+    protected function getCategoryRepository()
+    {
+        $repository = $this->manager->getRepository(
+            'Shopware\Models\Category\Category'
```
Comment (sbofirov) [inline Components/CategoryQuery/Sw41Query.php]: same here, ```Category::class```. Also we don't need variable ```$repository```, just return $this->manager->getRepository(...

File: Components/Utils/CountryCodeResolver.php [hunk 3]
```diff
@@ -18,15 +18,29 @@
  */
 class CountryCodeResolver
 {
-    /** @var \Shopware\Components\Model\ModelManager */
+    /**
+     * @var \Shopware\Components\Model\ModelManager
```
Comment (sbofirov) [inline Components/Utils/CountryCodeResolver.php]: Only class name or full namespace? in previous files we use only class name.
Need to decide what to use in the future

File: Components/Utils/CountryCodeResolver.php [hunk 4]
```diff
@@ -18,15 +18,29 @@
...
+     */
     protected $modelManager;
-    /** @var \Shopware\Models\Country\Repository */
+
+    /**
+     * @var \Shopware\Models\Country\Repository
```
Comment (sbofirov) [inline Components/Utils/CountryCodeResolver.php]: Same here

File: Components/Utils/CountryCodeResolver.php [hunk 5]
```diff
@@ -18,15 +18,29 @@
...
-    /** @var string */
+
+    /**
+     * @var string
+     */
     protected $default = 'DEU';
-    /** @var \Shopware\Models\Customer\Customer */
+
+    /**
+     * @var \Shopware\Models\Customer\Customer
```
Comment (sbofirov) [inline Components/Utils/CountryCodeResolver.php]: Same here

File: Components/Utils/CountryCodeResolver.php [hunk 6]
```diff
@@ -35,10 +49,10 @@ class CountryCodeResolver
      * @param null $countryId               CountryID - e.g. from session
      * @param string $default               Fallback
      */
-    public function __construct(ModelManager $modelManager, Customer $customer=null, $countryId=null, $default = 'DEU')
+    public function __construct(ModelManager $modelManager, Customer $customer = null, $countryId = null, $default = 'DEU')
     {
         $this->modelManager = $modelManager;
-        $this->default = 'DEU';
+        $this->default = $default;
```
Comment (sbofirov) [inline Components/Utils/CountryCodeResolver.php]: :+1:

File: Tests/Shopware/Connect/Sw41QueryTest.php [hunk 7]
```diff
@@ -1,131 +0,0 @@
...
- * file that was distributed with this source code.
- */
-
-namespace Tests\ShopwarePlugins\Connect\CategoryQuery;
-
-use Shopware\Connect\Struct\Product;
-use ShopwarePlugins\Connect\Components\ConnectFactory;
-use Tests\ShopwarePlugins\Connect\ConnectTestHelper;
-
-class Sw41QueryTest extends ConnectTestHelper
```
Comment (sbofirov) [inline Tests/Shopware/Connect/Sw41QueryTest.php]: Why this test was removed? it's still need or not ?

Comment (SimonBaeumer) [inline Tests/Shopware/Connect/Sw41QueryTest.php]: It tested a method which was removed in this pr beacuse it was not used in production code

File: Components/CategoryQuery/Sw41Query.php [hunk 8]
```diff
@@ -42,27 +46,25 @@ public function getRelevanceSorter()
...
      */
     protected function getArticleRepository()
     {
-        $repository = $this->manager->getRepository(
-            'Shopware\Models\Article\Article'
+        return $this->manager->getRepository(
+            Article::class
         );
 
         return $repository;
```
Comment (sbofirov) [inline Components/CategoryQuery/Sw41Query.php]: Forgot to remove :)

**Pull request merged**